doc/variableexplorer.rst

```diff
@@ -2,95 +2,213 @@
+|
 
+You will also find the option to remove all variables from the Variable Explorer and the option to search variables by its name or type, filtering according to the text entered in the search box.
 
+.. image:: images/variable_explorer/variable-explorer-search.gif
+   :align: center
+   :alt: Variable Explorer showing how to search variables
+
+|
+
+Finally you will find an option to refresh the Variable Explorer while running code in the IPython console.
+
+(MISSING GIF)
+
+================
+Advanced Options
+================
+
+You can also get an additional menu of options by right clicking on top of any of the variables. Through this menu, you can rename, remove or edit any variable. You can also make a copy of it by clicking the option "duplicate", which will open a dialog box in which you can write the name of the new copy of this variable.
+
+.. image:: images/variable_explorer/variable-explorer-duplicate.gif
+   :align: center
+   :alt: Variable Explorer showing duplicating a variable
+
+| 
+
+Besides, you can copy and paste the value of a variable, saving it in the Variable Explorer with any name that you choose. This allows you to change the type of the variable that you are pasting which turns out to be really useful, for example if you want to copy the elements of a list into an array.
+
+.. image:: images/variable_explorer/variable-explorer-copy-paste.gif
+   :align: center
+   :alt: Variable Explorer showing copying list into array
+
+| 
+
+Additionally, you can create a variable from zero directly in the Variable Explorer by clicking the option Insert, which allows you to write the key (which should be in quotation marks) and the value for the variable that you want to insert.
+
+.. image:: images/variable_explorer/variable-explorer-insert.gif
+   :align: center
+   :alt: Variable Explorer showing insertion of a new variable
+
+| 
+
+For lists and Numpy arrays, you will find more advanced options which include plotting or making a histogram using the data of the variable.
+
+.. image:: images/variable_explorer/variable-explorer-histogram.png
+   :align: center
+   :alt: Plot window showing a histogram, generated via the previous options
+
+|
+
+.. image:: images/variable_explorer/variable-explorer-plot.png
+   :align: center
+   :alt: Plot window showing a plot, generated via the previous options
+
+|
+
+You can save an array in disk too, in a file with npy extension, by clicking the option on this menu. 
+
+.. image:: images/variable_explorer/variable-explorer-contextmenu-array.png
+   :align: center
+   :alt: Context menu for an int array, with the Show image option selected
+
+|
+
+In the case of two-dimensional arrays, you can also show an image representing pixels using the numbers of the array as RGB values. For this, Spyder uses Matplotlib's colormaps, which can be easily changed according to your own preferences. To learn how to do this, you can go to "https://matplotlib.org/3.1.0/tutorials/colors/colormaps.html"
```

Comment:
```suggestion
For two-dimensional arrays, you can also display them as images, treating their values as RGB colors. For this, Spyder uses Matplotlib's colormaps, which can be `easily changed to match your preferences`_.

.. _easily changed to match your preferences: https://matplotlib.org/3.1.0/tutorials/colors/colormaps.html
```

If its using the values as RGB, doesn't it need to be three dimensional? It sounds like its just grayscale if its colored via Matplotlib's colormaps, not RGB.